### ISSUE TITLE
Running mean and running var use EMA?

### ISSUE THREAD

**Issue by zhaohui-yang** [fadel/pytorch_ema#13]
Thank you for your selfless sharing. In the author's application, EMA has been applied to all parameters that can be trained, but there are some statistics, such as the mean and variance of BN, that are not training parameters. When ema is copied back, this part of the statistics is adapted to the original parameters, not to the parameters after ema. I would like to ask if the author has ever experimented with ema for all state_dicts()? What is the experimental effect of smoothing variables in dict, not just trainable parameters?

**Comment by fadel** [comment]
An old issue, but for anyone browsing this and wondering: you should update the BatchNorm statistics after applying the EMA parameters (i.e., one pass over the training data), otherwise they would be wrong.

Unfortunately, BatchNorm is data-dependent and changing the model changes the data it sees, so the statistics kept by BatchNorm must be updated again.
